apps/ledger-live-desktop/src/renderer/components/PostOnboardingHub/PostOnboardingActionRow.tsx

```diff
@@ -29,29 +31,57 @@ const PostOnboardingActionRow: React.FC<Props> = props => {
     tagLabel,
     buttonLabelForAnalyticsEvent,
     completed,
+    getIsAlreadyCompleted,
     deviceModelId,
+    shouldCompleteOnStart,
   } = props;
   const { t } = useTranslation();
   const dispatch = useDispatch();
   const history = useHistory();
+  const recoverServices = useFeature("protectServicesDesktop");
+  const protectId = recoverServices?.params?.protectId ?? "protect-prod";
+
+  const completeAction = useCompleteActionCallback();
+  const [isActionCompleted, setIsActionCompleted] = useState(false);
+
+  const initIsActionCompleted = useCallback(async () => {
+    setIsActionCompleted(completed || !!(await getIsAlreadyCompleted?.({ protectId })));
+  }, [setIsActionCompleted, completed, getIsAlreadyCompleted, protectId]);
+
+  useEffect(() => {
+    initIsActionCompleted();
+  }, [initIsActionCompleted]);
```

Comment:
Do you really want to re-run the `init` if some dep changed ?

Reply:
It makes sense to recompute the isActionCompleted state if a prop changes for some reason

Reply:
Ok it just felt weird to call multiple times a function called init